packages/std/src/errors/recover_pubkey_error.rs

```diff
@@ -1,22 +1,24 @@
-#[cfg(not(target_arch = "wasm32"))]
+#[cfg(all(not(target_arch = "wasm32"), feature = "std"))]
 use cosmwasm_crypto::CryptoError;
 #[cfg(feature = "backtraces")]
 use std::backtrace::Backtrace;
-use std::fmt::Debug;
-use thiserror::Error;
 
-#[derive(Error, Debug)]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
```

Comment:
In no-std, `StdError: Error` will not be implemented, right?

I think it would be good to make some alternative implementation for no-std, so we can at least have `to_string()` and treat it like an error.